pymoto/modules/assembly.py

```diff
@@ -431,3 +431,104 @@ def _prepare(self, domain: DomainDefinition,
         if domain.dim == 2:
             D *= domain.element_size[2]
         self.element_matrix = D @ self.element_matrix
+
+
+class ElementAverage(ElementOperation):
+    r""" Determine average value in element of input nodal values
+
+    Input Signal:
+       - ``v``: Nodal vector of size ``(#dof per element * #nodes)``
+
+    Output Signal:
+       - ``v_el``: Elemental vector of size ``(#elements)``
+
+    Args:
+        domain: The domain defining element and nodal connectivity
+
+    Keyword Args:
+        ndof: Amount of dofs per node of nodal vector to average
+    """
+    def _prepare(self, domain: DomainDefinition, ndof = 1):
```

Comment:
I think `ndof` could be deduced when `response()` is called, as the input is `ndof * nnodes`. Do you think it would be possible to remove the argument `ndof`?

Reply:
Don't think so as `_prepare` of `ElementOperation` uses `self.element_matrix` to determine `ndof` and subsequently `self.dofconn`, so doing so would require changing `ElementOperation` significantly.

Reply:
I have upgraded the `ElementOperation` such that it repeats the element matrix if the size matches with the number of nodes instead of the number of dofs. This means the argument `ndof` is no longer required.